ts/src/xt.ts

```diff
@@ -0,0 +1,4239 @@
+        //                 "cr": "0.0015",
+        //                 "o": "22398.05",
+        //                 "l": "22323.72",
+        //                 "h": "22600.50",
+        //                 "c": "22432.05",
+        //                 "q": "7962.256931",
+        //                 "v": "178675209.47416856"
+        //             }
+        //         ]
+        //     }
+        //
+        // swap and future
+        //
+        //     {
+        //         "returnCode": 0,
+        //         "msgInfo": "success",
+        //         "error": null,
+        //         "result": {
+        //             "t": 1678172848572,
+        //             "s": "btc_usdt",
+        //             "c": "22415.5",
+        //             "h": "22590.0",
+        //             "l": "22310.0",
+        //             "a": "623654031",
+        //             "v": "1399166074.31675",
+        //             "o": "22381.5",
+        //             "r": "0.0015",
+        //             "i": "22424.5",
+        //             "m": "22416.5",
+        //             "bp": "22415",
+        //             "ap": "22415.5"
+        //         }
+        //     }
+        //
+        const ticker = this.safeValue (response, 'result');
+        if (market['spot']) {
+            return this.parseTicker (ticker[0], market);
+        }
+        return this.parseTicker (ticker, market);
+    }
+
+    async fetchTickers (symbols: string[] = undefined, params = {}) {
+        /**
+         * @method
+         * @name xt#fetchTickers
+         * @description fetches price tickers for multiple markets, statistical calculations with the information calculated over the past 24 hours each market
+         * @see https://doc.xt.com/#market10ticker24h
+         * @see https://doc.xt.com/#futures_quotesgetAggTickers
+         * @param {[string]|undefined} symbols unified symbols of the markets to fetch the ticker for, all market tickers are returned if not assigned
+         * @param {object} params extra parameters specific to the xt api endpoint
+         * @returns {object} an array of [ticker structures]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        await this.loadMarkets ();
+        let market = undefined;
+        if (symbols !== undefined) {
+            symbols = this.marketSymbols (symbols);
+            market = this.market (symbols[0]);
+        }
+        const request = {};
+        let subType = undefined;
+        [ subType, params ] = this.handleSubTypeAndParams ('fetchTickers', market, params);
+        let response = undefined;
+        if (market['spot']) {
+            request['symbols'] = symbols;
+            response = await this.publicSpotGetTicker24h (this.extend (request, params));
+        } else {
+            if (symbols !== undefined) {
+                throw new NotSupported (this.id + ' the symbols argument is not supported for swap and future markets');
```

Comment:
@Dan-krm we can always do the request without any `symbols` and filter it afterwards